src/ragas/config.py

```diff
@@ -1,27 +1,32 @@
+from __future__ import annotations
+
 import typing as t
 
 from pydantic import BaseModel, Field
 
-from ragas.embeddings import BaseRagasEmbeddings
-from ragas.llms import BaseRagasLLM
+from ragas.embeddings import BaseRagasEmbeddings, embedding_factory
+from ragas.llms import BaseRagasLLM, llm_factory
 from ragas.losses import Loss
 from ragas.optimizers import GeneticOptimizer, Optimizer
 
 DEFAULT_OPTIMIZER_CONFIG = {"max_steps": 100}
 
 
 class DemonstrationConfig(BaseModel):
+    embedding: BaseRagasEmbeddings = Field(default_factory=embedding_factory)
```

Comment:
Same idea here as model

Reply:
got it

Reply:
but can't have that here because `train()` is in `MetricWithLLM` so don't have access to embedding

I think we should make LLM explicit, force the developer to choose it